### ISSUE TITLE
Can't like comments on certain news feed posts & Posts without attachment are not being shown on user/group/event profile.

### ISSUE THREAD

**Issue by jonathanbloom** [SocialEngine/phpv4-issues#7]
We have an issue liking comments on Link posts on newsfeed in 4.8.7.

Log:
Hi, we checked this and found problem in code, you would need to report this to SE support as it is bug with code, error is:

```
Stack trace:
#0 /home/USERNAME/public_html/application/libraries/Zend/Controller/Action.php(516): Activity_IndexController->likeAction()
#1 /home/USERNAME/public_html/application/libraries/Zend/Controller/Dispatcher/Standard.php(308): Zend_Controller_Action->dispatch('likeAction')
#2 /home/USERNAME/public_html/application/libraries/Zend/Controller/Front.php(954): Zend_Controller_Dispatcher_Standard->dispatch(Object(Zend_Controller_Request_Http), Object(Zend_Controller_Response_Http))
#3 /home/USERNAME/public_html/application/modules/Core/Bootstrap.php(80): Zend_Controller_Front->dispatch()
#4 /home/USERNAME/public_html/application/libraries/Engine/Application.php(160): Core_Bootstrap->run()
#5 /home/USERNAME/public_html/application/index.php(205): Engine_Application->run()
#6 /home/USERNAME/public_html/index.php(24): include('/home/USERNAME/p...')
#7 {main}
```

**Comment by smazurov** [comment]
Hi @jonathanbloom. The stack trace is actually missing the error, it contains where its coming from, but not the error itself, could you edit the post to include that?

P.S. I took a bit of liberty and used some [markdown formatting](https://guides.github.com/features/mastering-markdown/) as well as removed the username in one instance where it was still visible.

**Comment by jonathanbloom** [comment]
Sorry about that,

`2015-02-18T02:14:56+00:00 CRIT (2): 
Error Code: e6ce50
exception 'Engine_Exception' with message 'This user is not allowed to like this item' in /home/USERNAME/public_html/application/modules/Activity/controllers/IndexController.php:401
Stack trace:
#0 /home/USERNAME/public_html/application/libraries/Zend/Controller/Action.php(516): Activity_IndexController->likeAction()
#1 /home/USERNAME/public_html/application/libraries/Zend/Controller/Dispatcher/Standard.php(308): Zend_Controller_Action->dispatch('likeAction')
#2 /home/USERNAME/public_html/application/libraries/Zend/Controller/Front.php(954): Zend_Controller_Dispatcher_Standard->dispatch(Object(Zend_Controller_Request_Http), Object(Zend_Controller_Response_Http))
#3 /home/USERNAME/public_html/application/modules/Core/Bootstrap.php(80): Zend_Controller_Front->dispatch()
#4 /home/USERNAME/public_html/application/libraries/Engine/Application.php(160): Core_Bootstrap->run()
#5 /home/USERNAME/public_html/application/index.php(205): Engine_Application->run()
#6 /home/USERNAME/public_html/index.php(24): include('/home/tniw593/p...')
#7 {main}`

From our testing it seems to happen on comments on posts for Links.

**Comment by shivani2910** [comment]
Hi @jonathanbloom, 

We've been able to reproduce this issue. Our development team is working on this and we'll post a solution for this issue in this thread at the earliest.

Best Regards,
Shivani

**Comment by shivani2910** [comment]
Hi @jonathanbloom, 

Sorry for the delay in responding to you.

Please find below the solution to this issue. Below is the list of files that you need to replace downloading the corresponding linked files we've added:

[application/modules/Activity/controllers/IndexController.php] ( https://gist.githubusercontent.com/shivani2910/9076fa4ada9c460c16a1/raw/8c8166694dddd367ea3310dcda4ca171b170c9b3/IndexController.php )
[application/modules/Activity/views/scripts/_activityComments.tpl] ( https://gist.githubusercontent.com/shivani2910/9076fa4ada9c460c16a1/raw/6e6938651c6e6c9c20487179878b6173d6425fe9/_activityComments.tpl )
[application/modules/Activity/views/scripts/_activityText.tpl] ( https://gist.githubusercontent.com/shivani2910/9076fa4ada9c460c16a1/raw/fee378e5711db5fcee2efbe417a532a2888937ee/_activityText.tpl )
[application/modules/Activity/Model/Action.php] ( https://gist.githubusercontent.com/shivani2910/9076fa4ada9c460c16a1/raw/72fc284359415bbf50aac5cd8c90c47795eda8d5/Action.php )
[application/modules/Core/Model/Link.php] ( https://gist.githubusercontent.com/shivani2910/9076fa4ada9c460c16a1/raw/d3cb3953711f16c0df1158e39a7c5b9cfae0e018/Link.php )

Once you'll replace these 5 files on your server with the given files thee your problem will be resolved. If the issue still persists then please drop us a line.

Best Regards,
Shivani

**Comment by jonathanbloom** [comment]
Thanks Shivani, it is working now!

**Comment by jonathanbloom** [comment]
After installing the fixes, some of my link posts show as 0 comments.

**Comment by jonathanbloom** [comment]
Correction, they lost comments it seems.

**Comment by jonathanbloom** [comment]
And also this post had likes and now shows as 0 likes.

**Comment by smazurov** [comment]
fixed in [4.8.8](http://blog.socialengine.com/2015/03/23/socialengine-php-4-8-8-released/)
